lib/Infer/AliveDriver.cpp

```diff
@@ -625,10 +634,31 @@ bool souper::AliveDriver::translateAndCache(const souper::Inst *I,
 
     case souper::Inst::Phi: {
       if (I->Ops.size() != 1) {
-        assert(false && "Phi with muliple arguments unimplemented");
-        return false;
+        auto FreeVar = Builder.undef(getType(I->Width),
+                       "%" + std::to_string(InstNumbers++));
+
+        auto PhiConstraint = Builder.iCmp(getType(1),
+             "%" + std::to_string(InstNumbers++),
+             IR::ICmp::EQ, FreeVar, ExprCache[I->Ops[0]]);
+
+        for (int i = 1; i < I->Ops.size(); ++i) {
+          auto Current = Builder.iCmp(getType(1),
+             "%" + std::to_string(InstNumbers++),
+             IR::ICmp::EQ, FreeVar, ExprCache[I->Ops[0]]);
```

Comment:
The last argument should be ExprCache[I->Ops[i]]

Reply:
Are you sure that the index 0 of I->Ops is the value that comes from predecessor 0 rather than the block id? I forgot, let me check.

Reply:
Block id is a separate member in Inst.
Fixed Ops[i]